storage/pinning/pinning.go

```diff
@@ -60,12 +60,36 @@ func (prov *Provider) Write(ctx context.Context, data []byte) (string, error) {
 		return "", err
 	}
 
-	req, err := http.NewRequest(http.MethodPost, prov.host+"/content/add-car", car)
+	// use a pipe to stream request data
+	pr, pw := io.Pipe()
+	// write form data to the pipe writer
+	mw := multipart.NewWriter(pw)
+	// pipe reader will block until content is written to pipe writer
+	// this is how the data is able to be streamed in a routine
+	go func() {
+		defer mw.Close()
```

Comment:
The writer is never properly closed. This should fix it.

```suggestion
                 // create a form file for the multipart data
		ff, err := mw.CreateFormFile("path", strings.TrimPrefix(fpath, "/ipfs/"))
		if err != nil {
			pw.CloseWithError(err) //nolint:errcheck
			return
		}
		// copy the input data to the form file
		_, err = io.Copy(ff, car)
		if err != nil {
			pw.CloseWithError(err) //nolint:errcheck
			return
		}
		
		err = mw.Close()
		pw.CloseWithError(err) //nolint:errcheck
```